code/modules/power/singularity/generator.dm

```diff
@@ -13,7 +13,8 @@
 /obj/machinery/the_singularitygen/process()
 	var/turf/T = get_turf(src)
 	if(src.energy >= 200)
-		new creation_type(T, 50)
+		var/obj/singularity/S = new creation_type(T, 50)

```

Comment:
tesla ball is a child of hte singulo anyway. this doesn't really matter, you can just do var/S if you want